assets/kopijs.js

```diff
@@ -20,67 +20,108 @@
-      + '</div>';
+      // Render the (random) ice(s)
+      var randDeg = function(){
+        // Returns -90 to 90 deg
+        return (Math.random()*180).toFixed() - 90;
+      };
+      var randLeft = function(){
+        // Returns 8 to 26
+        return (Math.random()*18 + 8).toFixed();
+      };
+      iceHTML = '<div class="ice-blocks">'
+          + '<div class="ice" style="transform: rotate(' + randDeg() + 'deg); -webkit-transform: rotate(' + randDeg() + 'deg); left: ' + randLeft() + 'px"></div>'
+          + '<div class="ice" style="transform: rotate(' + randDeg() + 'deg); -webkit-transform: rotate(' + randDeg() + 'deg); left: ' + randLeft() + 'px"></div>'
+          + '<div class="ice" style="transform: rotate(' + randDeg() + 'deg); -webkit-transform: rotate(' + randDeg() + 'deg); left: ' + randLeft() + 'px"></div>'
+        + '</div>';
 
-    var state = ingredients.state;
-    ingredients.state = 'lukewarm';
-    var lukewarmName = Kopi.stringify(ingredients);
-    ingredients.state = 'iced';
-    var icedName = Kopi.stringify(ingredients);
+      var state = ingredients.state;
+      ingredients.state = 'lukewarm';
+      var lukewarmName = Kopi.stringify(ingredients);
+      ingredients.state = 'iced';
+      var icedName = Kopi.stringify(ingredients);
 
-    // Finally render the whole Kopi in its full glory
-    var html = '<div class="kopi kopi-' + state + '">'
-        + sugarHTML
-        + '<div class="cup">'
-          + '<div class="ingredients">' + internalIngredientsHTML + '</div>'
-          + iceHTML
+      // Finally render the whole Kopi in its full glory
+      var html = '<div class="kopi kopi-' + state + '">'
+          + sugarHTML
+          + '<div class="cup">'
+            + '<div class="ingredients">' + internalIngredientsHTML + '</div>'
+            + iceHTML
+          + '</div>'
+          + '<div class="plate"></div>'
         + '</div>'
-        + '<div class="plate"></div>'
-      + '</div>'
-      + '<div class="kopi-name">'
-        + '<span class="kopi-warm-name">' + name + '</span>'
-        + '<span class="kopi-lukewarm-name">' + lukewarmName + '</span>'
-        + '<span class="kopi-iced-name">' + icedName + '</span>'
-      + '</div>';
-    $kopi.innerHTML = html;
+        + '<div class="kopi-name">'
+          + '<span class="kopi-warm-name">' + name + '</span>'
+          + '<span class="kopi-lukewarm-name">' + lukewarmName + '</span>'
+          + '<span class="kopi-iced-name">' + icedName + '</span>'
+        + '</div>';
+      $kopi.innerHTML = html;
+  }
+
+  //Make your own kopi
+  var $recipe = document.querySelectorAll('#recipe > input');
+  var $ownKopi = document.getElementById('your-kopi');
+  $ownKopi.className = 'warm';
+  makeKopi($ownKopi, null, {water: 0.4, coffee: 0.4, condensed_milk: 0.2, state: "warm"});
+  //indication of amount
+  for(var i = 0; i<5; i++){
+      $recipe[i].addEventListener('input', function(){
```

Comment:
Perhaps try event delegate the events instead of binding the `input` event one by one for every single `input` 🤔